src/UbiHttp.cpp

```diff
@@ -0,0 +1,405 @@
+    if (_client_https_ubi.available()) {
+      String line = _client_https_ubi.readStringUntil('\n');
+      if (line == "\r") {
+        break;
+      }
+    }
+  }
+
+  free(message);
+  free(path);
+
+  double value = _parseServerAnswer();
+
+  _client_https_ubi.flush();
+  _client_https_ubi.stop();
+  return value;
+}
+
+double UbiHTTP::_parseServerAnswer() {
+  /**
+   * @param _charResponseLength 3 is the maximun amount of digits for the length
+   * to have
+   */
+  char *_charLength = (char *)malloc(sizeof(char) * 3);
+
+  _parsePartialServerAnswer(_charLength);
+  /**
+   * The server respond the length of the value in HEX so it has to be converted
+   * to DEC
+   * */
+  uint8_t length = UbiUtils::hexadecimalToDecimal(_charLength);
+  char *_charValue = (char *)malloc(sizeof(char) * length + 1);
+
+  _parsePartialServerAnswer(_charValue);
+
+  double value = strtof(_charValue, NULL);
+
+  free(_charLength);
+  free(_charValue);
+
+  return value;
+}
+
+/**
+ * @brief Calculate the lenght of the request line to be send over HTTP to the
+ * server
+ *
+ * @param path address of the endpoint to gather the data
+ * @return uint16_t  Lenght of the request line
+ */
+uint16_t UbiHTTP::_requestLineLength(char *path) {
+  uint16_t endpointLength = strlen(
+                                "GET  HTTP/1.1\r\nHost: \r\nX-Auth-Token: "
+                                "\r\nUser-Agent: \r\nContent-Type: "
+                                "application/json\r\nConnection: close\r\n\r\n") +
+                            strlen(path) + strlen(_host) + strlen(_token) + strlen(_user_agent);
+  return endpointLength;
+}
+
+/**
+ * @brief Calculate the lenght of the path to be send over HTTP to the server
+ *
+ * @param device_label device label of the device
+ * @param variable_label variable label to be updated or fetched
+ * @return uint16_t  Lenght of the endpoint path
+ */
+uint16_t UbiHTTP::_pathLength(const char *device_label, const char *variable_label) {
+  uint16_t endpointLength = strlen("/api/v1.6/devices///lv") + strlen(device_label) + strlen(variable_label);
+  return endpointLength;
+}
+
+/**
+ * Reconnects to the server
+ * @return true once the host answer buffer length is greater than zero,
+ *         false if timeout is reached.
+ */
+
+bool UbiHTTP::reconnect(const char *host, const int port) {
+  uint8_t attempts = 0;
+  bool connected = false;
+  while (!connected && attempts < _maxReconnectAttempts) {
+    if (_debug) {
+      Serial.print(F("Trying to connect to "));
+      Serial.print(host);
+      Serial.print(F(" , attempt number: "));
+      Serial.println(attempts);
+    }
+    attempts += 1;
+    connected = _client_https_ubi.connect(host, port);
+    delay(1000);
+  }
+  if (connected) {
+    if (_debug) {
+      Serial.println("Connected");
+    }
+  } else {
+    if (_debug) {
+      Serial.println("Could not connect to the server, please check your network and your firewall rules");
```

Comment:
Missing PROGMEM allocation when printing the message `Serial.println(F("Message......."))`